src/core/constraint_template.jl

```diff
@@ -182,58 +182,58 @@ function constraint_tp_flow_losses(pm::_PMs.GenericPowerModel, i::Int; nw::Int=p
 end
 
 
 ""
-function constraint_tp_storage_exchange(pm::_PMs.GenericPowerModel, i::Int; nw::Int=pm.cnw)
+function constraint_mc_storage_exchange(pm::_PMs.AbstractPowerModel, i::Int; nw::Int=pm.cnw)
     storage = _PMs.ref(pm, nw, :storage, i)
 
     _PMs.constraint_storage_complementarity_nl(pm, nw, i)
-    constraint_tp_storage_loss(pm, nw, i, storage["storage_bus"], storage["r"], storage["x"], storage["standby_loss"])
+    _PMs.constraint_storage_loss(pm, nw, i, storage["storage_bus"], _PMs.conductor_ids(pm), storage["r"], storage["x"], storage["p_loss"], storage["q_loss"])
 end
 
 
 ""
-function constraint_tp_trans(pm::_PMs.GenericPowerModel, i::Int; nw::Int=pm.cnw)
+function constraint_mc_trans(pm::_PMs.AbstractPowerModel, i::Int; nw::Int=pm.cnw)
     if _PMs.ref(pm, pm.cnw, :conductors)!=3
         Memento.error(_LOGGER, "Transformers only work with networks with three conductors.")
     end
     (Tv_fr,Tv_im,Ti_fr,Ti_im,Cv_to) = _calc_tp_trans_Tvi(pm, i)
     f_bus = _PMs.ref(pm, :trans, i)["f_bus"]
     t_bus = _PMs.ref(pm, :trans, i)["t_bus"]
     tm = _PMs.ref(pm, :trans, i)["tm"]
-    constraint_tp_trans_voltage(pm, nw, i, f_bus, t_bus, tm, Tv_fr, Tv_im, Cv_to)
+    constraint_mc_trans_voltage(pm, nw, i, f_bus, t_bus, tm, Tv_fr, Tv_im, Cv_to)
     f_idx = (i, f_bus, t_bus)
     t_idx = (i, t_bus, f_bus)
-    constraint_tp_trans_flow(pm, nw, i, f_bus, t_bus, f_idx, t_idx, tm, Ti_fr, Ti_im, Cv_to)
+    constraint_mc_trans_flow(pm, nw, i, f_bus, t_bus, f_idx, t_idx, tm, Ti_fr, Ti_im, Cv_to)
 end
 
 
 ""
-function constraint_tp_oltc(pm::_PMs.GenericPowerModel, i::Int; nw::Int=pm.cnw)
+function constraint_mc_oltc(pm::_PMs.AbstractPowerModel, i::Int; nw::Int=pm.cnw)
     if _PMs.ref(pm, pm.cnw, :conductors)!=3
         Memento.error(_LOGGER, "Transformers only work with networks with three conductors.")
     end
     (Tv_fr,Tv_im,Ti_fr,Ti_im,Cv_to) = _calc_tp_trans_Tvi(pm, i)
     trans = _PMs.ref(pm, :trans, i)
     f_bus = trans["f_bus"]
     t_bus = trans["t_bus"]
-    constraint_tp_oltc_voltage(pm, nw, i, f_bus, t_bus, Tv_fr, Tv_im, Cv_to)
+    constraint_mc_oltc_voltage(pm, nw, i, f_bus, t_bus, Tv_fr, Tv_im, Cv_to)
     f_idx = (i, f_bus, t_bus)
     t_idx = (i, t_bus, f_bus)
-    constraint_tp_oltc_flow(pm, nw, i, f_bus, t_bus, f_idx, t_idx, Ti_fr, Ti_im, Cv_to)
+    constraint_mc_oltc_flow(pm, nw, i, f_bus, t_bus, f_idx, t_idx, Ti_fr, Ti_im, Cv_to)
     # fix the taps with a constraint which are not free
     trans = _PMs.ref(pm, :trans, i)
     fixed = trans["fixed"]
     tm = trans["tm"]
-    constraint_tp_oltc_tap_fix(pm, i, fixed, tm)
+    constraint_mc_oltc_tap_fix(pm, i, fixed, tm)
 end
 
 
 "KCL including transformer arcs."
-function constraint_tp_power_balance_shunt_trans(pm::_PMs.GenericPowerModel, i::Int; nw::Int=pm.cnw, cnd::Int=pm.ccnd)
+function constraint_mc_power_balance_shunt_trans(pm::_PMs.AbstractPowerModel, i::Int; nw::Int=pm.cnw, cnd::Int=pm.ccnd)
```

Comment:
PowerModels v0.13 proposed a method for a "generic" power balance constraint.  We should post an issue to see if a similar design can be used in this package.

Reply:
Definitely I like the idea of making this function more generic, instead of having so many different variations for storage, loads, transformers, etc.